src/overlays/actors/ovl_Bg_Hakugin_Elvpole/z_bg_hakugin_elvpole.c

```diff
@@ -30,17 +30,111 @@ const ActorInit Bg_Hakugin_Elvpole_InitVars = {
+            this->dyna.actor.world.pos.y += 4.0f;
+            sp28 = true;
+        } else if (this->unk_15E > 0) {
+            if (this->unk_15E == 100) {
+                Actor_PlaySfxAtPos(&this->dyna.actor, NA_SE_EV_STONEDOOR_STOP);
+            }
+            this->dyna.actor.world.pos.y = this->dyna.actor.home.pos.y - 120.0f;
+            // FAKE
+            if (1) {}
+            this->unk_15E--;
+        }
+    } else if (this->unk_15E == 0) {
+        if ((this->dyna.actor.world.pos.y - this->dyna.actor.home.pos.y) > -320.0f) {
+            this->dyna.actor.world.pos.y -= 4.0f;
+            sp28 = true;
+        } else {
+            this->dyna.actor.world.pos.y = this->dyna.actor.home.pos.y - 320.0f;
+            Flags_UnsetSwitch(play, BGHAKUGINELVPOLE_GET_SWITCHFLAG(&this->dyna.actor));
+            this->unk_15E = -1;
+            Actor_PlaySfxAtPos(&this->dyna.actor, NA_SE_EV_STONEDOOR_STOP);
+        }
+    } else {
+        if (((BGHAKUGINELVPOLE_GET_SWITCHFLAG(&this->dyna.actor)) != 0x7F) &&
+            (Flags_GetSwitch(play, BGHAKUGINELVPOLE_GET_SWITCHFLAG(&this->dyna.actor)))) {
+            this->unk_15E = 0x64;
+            this->unk_160 = true;
+        }
+    }
+    if ((sp28) || ((this->unk_15C & 7) != 0)) {
+        if (this->unk_15C & 1) {
+            var_fv1 = 1.0f;
+        } else {
+            var_fv1 = -1.0f;
+        }
+        this->unk_15C++;
+        this->dyna.actor.world.pos.x = (Math_SinS(this->unk_15C * 0x2000) * var_fv1) + this->dyna.actor.home.pos.x;
+        this->dyna.actor.world.pos.z = (Math_CosS(this->unk_15C * 0x2000) * var_fv1) + this->dyna.actor.home.pos.z;
+        func_800B9010(&this->dyna.actor, 0x2103U);
+    } else {
+        this->unk_15C = 0;
+    }
+    if (this->unk_160) {
+        if (this->dyna.actor.cutscene == -1) {
+            this->unk_160 = false;
+            return;
+        }
+        if (ActorCutscene_GetCanPlayNext(this->dyna.actor.cutscene)) {
+            ActorCutscene_StartAndSetUnkLinkFields(this->dyna.actor.cutscene, &this->dyna.actor);
+            this->unk_160 = false;
+            return;
+        }
+        ActorCutscene_SetIntentToPlay(this->dyna.actor.cutscene);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Hakugin_Elvpole/BgHakuginElvpole_Draw.s")
+// #pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Hakugin_Elvpole/BgHakuginElvpole_Update.s")
+void BgHakuginElvpole_Update(Actor* thisx, PlayState* play) {
+    BgHakuginElvpole* this = THIS;
+
+    this->actionFunc(this, play);
+}
+
+// #pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Hakugin_Elvpole/BgHakuginElvpole_Draw.s")
+void BgHakuginElvpole_Draw(Actor* thisx, PlayState* play) {
+    OPEN_DISPS(play->state.gfxCtx);
+    gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(play->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    func_8012C28C(play->state.gfxCtx);
+    gSPDisplayList(POLY_OPA_DISP++, object_hakugin_obj_DL_00ACB8);
+    CLOSE_DISPS(play->state.gfxCtx);
```

Comment:
```suggestion
    OPEN_DISPS(play->state.gfxCtx);

    gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(play->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
    func_8012C28C(play->state.gfxCtx);
    gSPDisplayList(POLY_OPA_DISP++, object_hakugin_obj_DL_00ACB8);

    CLOSE_DISPS(play->state.gfxCtx);
```
Not required, but helps to make the inside stick out a bit.